rpc/websocket_server.go

```diff
@@ -186,19 +246,26 @@ func (w *WebSocketServer) MessageCallback(vm VM) pubsub.Callback {
 
 			// Submit will remove from [txWaiters] if it is not added
 			txID := tx.ID()
-			if err := vm.Submit(ctx, false, []*chain.Transaction{tx})[0]; err != nil {
-				log.Error("failed to submit tx",
+			if err := w.vm.Submit(ctx, false, []*chain.Transaction{tx})[0]; err != nil {
+				w.logger.Error("failed to submit tx",
 					zap.Stringer("txID", txID),
 					zap.Error(err),
 				)
 				return
 			}
-			log.Debug("submitted tx", zap.Stringer("id", txID))
+			w.logger.Debug("submitted tx", zap.Stringer("id", txID))
 		default:
-			log.Error("unexpected message type",
+			w.logger.Error("unexpected message type",
 				zap.Int("len", len(msgBytes)),
 				zap.Uint8("mode", msgBytes[0]),
 			)
 		}
 	}
 }
+
+func (w *WebSocketServer) Initialize(vm VM) {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+
+	w.vm = vm
+}
```

Comment:
Is this needed? It seems that it's only called from the integration tests

Reply:
Was needed previously to fix the circular dependency issue but with the new option pattern we're doing this can be removed.